docs/ilm.asciidoc

```diff
@@ -0,0 +1,190 @@
+|=======================================================================
+|Event type |Hot |Warm
+|Errors & Spans
+|Rollover: `max_size: 50gb`, `max_age: 1 day`
+`priority: 100`
+|`min_age: 7 days`
+readonly
+`priority: 50`
+|Transactions & Metrics
+|Rollover: `max_size: 50gb`, `max_age: 7 days`
+`priority: 100`
+|`min_age: 31 days`
+readonly
+`priority: 50`
+|=======================================================================
+
+// OLD INFO, MAYBE DELETE 
+// *******
+
+**Error & Span events**
+_14 days to look at data in a very fast way_
+HOT
+* A new index is created every 50gb or 1 day (whichever comes first). This makes it very easy to delete indices day by day, and ensures no index gets too large.
+* After seven days, hot indices are moved to the warm phase.
+WARM
+* Warm indices are read-only.
+* After another seven days, warm indices are moved to the cold phase.
+COLD
+* Cold indices move data out of memory to clean up memory usage. Data can still be queried, albeit slowly.
+DELETE
+* By default, no delete phase is defined. Create a custom delete policy using the UI management section.
+
+**Transactions & Metrics**
+_Compare the current month with your last month without any performance reduction_
+HOT
+* A new index is created every 50gb or 7 days (whichever comes first). This makes it very easy to delete indices week by week, and ensures no index gets too large.
+* After one month (31 days), hot indices are moved to the warm phase.
+WARM
+* Warm indices are read-only.
+* After two more months (62 days), warm indices are moved to the cold phase.
+COLD
+* Cold indices move data out of memory to clean up memory usage. Data can still be queried, albeit slowly.
+DELETE
+* By default, no delete phase is defined. Create a custom delete policy using the UI management section.
+
+// *******
+// END OLD INFO
+
+[float]
+[[ilm-setup]]
+=== ILM setup
+
+To set up index lifecycle management, set `ilm.enabled` to `true` in apm-server.yml.
+ILM can only be enabled for `output.elasticsearch`.
+When enabled, configurations defined for `output.elasticsearch.index` and `output.elasticsearch.indices` will be ignored.
+ILM clears all index settings and sets up a predefined set of indices and policies that the user cannot initially configure.
+
+It is recommended to setup index lifecycle management (ILM) before starting APM Server.
+This prevents indices from being created before policies are changed, and allows policies to be applied to indices at creation time.
+It also excludes setup from the ingestion process, which allows you to ensure ILM is setup correctly before using APM.
+
+Index lifecycle management will manage an index based on its defined policy.
+Run the <<setup-command>> with the ` --index-management` flag to set up the default ILM policy:
+
+[source,js]
+-----------------------
+./apm-server setup --index-management
+-----------------------
+// CONSOLE
+
+//Policies only need to be created once, and will persist through version upgrades.
```

Comment:
Policies are versioned, so while they are persisted, they need to be setup for every version.